### DISCUSSION TITLE
pocketbase  TLS handshake error

### DISCUSSION THREAD

**Discussion by bikedel** [pocketbase/pocketbase#896] [Q&A]
Hello

I am on a new VPS
Centos8
Apache
SSL 

serving on http is all good

when I try serve with https I get errors

2022/11/02 17:14:21 http: TLS handshake error from 197.245.56.17:60699: acme/autocert: unable to satisfy "https://acme-v02.api.letsencrypt.org/acme/authz-v3/171534314487" for domain "42birds.co.za": no viable challenge type found

any help would be appreciated and thanks for the great work!

**Comment by ganigeorgiev** [comment]
Make sure that your domain is publicly accessible and if you are behind a cdn (eg. cloudflare) it allows both HTTP and HTTPS.

Running:

```sh
curl -Ii  42birds.co.za

HTTP/1.1 302 Found
Date: Wed, 02 Nov 2022 12:36:40 GMT
Server: Apache/2.4.37 (centos) OpenSSL/1.1.1k
Location: https://42birds.co.za/
Content-Type: text/html; charset=iso-8859-1
```

The above shows that you already have an issued certificate and redirect to HTTPS.

How are you deploying PocketBase? Are you using Apache as reverse proxy?

If PocketBase is behind a reverse proxy that already has something like certbot or similar to take care for the certificate, then you don't need the `--https` option.

If you are still not able to resolve your issue, please provide a little more info on your setup (including your apache proxy config) and the commands you are running.

If you haven't already, you can also check the [Going to production guide](https://pocketbase.io/docs/going-to-production/).

**Comment by bikedel** [comment]
thanks for the fast response

I have opened port 4000 in firewall.  No reverse proxy setup. out the box apache with normal ports 80 and 443.

 ./pocketbase serve --https 42birds.co.za:4000

**Reply by ganigeorgiev** [reply]
There seems to be a misunderstanding how PocketBase works. When you use the `--https` flag we issue a "Let's encrypt certificate".

But you can't issue a certificate on port different than 443 - https://community.letsencrypt.org/t/port-4434-instead-of-443/61349 (_actually you can, but not only browsers will not understand it, you'll also need some special server setup to handle the [domain challenges](https://letsencrypt.org/docs/challenge-types/)_).

---

PocketBase doesn't need Apache or anything else to work. If you are on clean linux vm and don't have other applications running, you can execute:

```sh
./pocketbase serve --http="yourdomain.com:80" --https="yourdomain.com:443"
```
And that's all.

---

If you want to use Apache (eg. you have multiple public facing applications on the same server), then you have to setup a reverse proxy.
I'm not very familiar with Apache, but you can check the available directives in https://httpd.apache.org/docs/2.4/howto/reverse_proxy.html. You could also refer to the nginx example in https://pocketbase.io/docs/going-to-production/#deploy-behind-reverse-proxy (it should be the same more-or-less).

Once you've setup the reverse proxy, you can start PocketBase on **127.0.0.1** (or localhost, 0.0.0.0, etc.) with the configured port you have in your proxy config, eg.:

```sh
./pocketbase serve --http="127.0.0.1:4000"
```

**Comment by bikedel** [comment]
Thanks for the quick reply and assistance.

I now have it running on port 443.

I wil look into the reverse poxy ….
